pym/bob/cmds/build.py

```diff
@@ -889,9 +892,18 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
             # Can we theoretically download the result? Exclude packages that
             # provide host tools when not building in a sandbox. Try to
             # determine a build-id for all other artifacts.
+            #
+            # Create build-id for shared packages, but don't download them
+            canLinkShared = False
+            canDownload = False
             if packageStep.doesProvideTools() and (packageStep.getSandbox() is None):
```

Comment:
I think the logic here (and on more locations below) should be

```python
if packageStep.isPackageRelocatable() or (packageStep.getSandbox() is not None):
```

The rationale is the following:
* inside the sandbox all paths are stable -> we never relocate
* outside the sandbox the package must be relocatable to work correctly
* non-relocatable packages are never up- or downloaded -> should never be a problem